### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by VitalikVoicix**
Make new branch from development for long-suffering categories

### CODE REVIEW

File: modules/category/category.model.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
 const mongoose = require('mongoose');
-const Product = require('../product/product.model');
```
Comment (moran711) [inline modules/category/category.model.js:2]: Please remove `isMain `and `subcategories `fields from category model.

File: modules/category/category.service.js [hunk 2]
```diff
@@ -26,61 +26,73 @@ class CategoryService {
...
+
+    if (!upload || !Object.keys(upload).length) {
+      return await Category.findByIdAndUpdate(id, category, { new: true });
     }
-    return await Category.findByIdAndUpdate(id, category, {
-      new: true,
-    });
+    const uploadResult = await uploadFiles([upload]);
+
+    const uploadResults = await uploadResult[0];
```
Comment (moran711) [inline modules/category/category.service.js:44]: If we pass more than one image it won't add it to category result.

**Review by moran711 — APPROVED** [review]

File: modules/category/category.model.js (hunk 1 above)
Comment (VitalikVoicix) [reply modules/category/category.model.js:2]: allready remove it

File: modules/category/category.service.js (hunk 2 above)
Comment (VitalikVoicix) [reply modules/category/category.service.js:44]: why do we need  pass  more than one image?

**Review by vasylshpak — APPROVED** [review]

**Review by Iryna-Bzdyr — APPROVED** [review]

**Review by nadiyafomenko — APPROVED** [review]

**Pull request merged**